python/cudf/cudf/_lib/CMakeLists.txt

```diff
@@ -89,6 +89,19 @@ foreach(target IN LISTS targets_using_dlpack)
   target_include_directories(${target} PRIVATE "${DLPACK_INCLUDE_DIR}")
 endforeach()
 
+find_package(Python 3.8 REQUIRED COMPONENTS Interpreter)
```

Comment:
This change was necessary because of the change of placement of header files in arrow-10:
```
arrow-9:

/nvme/0/pgali/envs/cudfdev/include/arrow/python/platform.h




arrow-10:

/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/pyarrow/src/arrow/python/platform.h
```

Reply:
Was this an intended change of pyarrow or was this a accidental change / bug? We should open an issue if this wasnt intended

Reply:
This was an intended change, all the python headers are removed from `arrow-cpp` and it is now being shipped as `libarrow` and the python headers have moved to its own `pyarrow` package.

Reply:
Right but why would the location of the headers change? They can still be part of `pyarrow` but exist in the old location

Reply:
These actually never existed in `pyarrow` before. Only existed in `arrow-cpp`, with `10.x` the files have been moved from `arrow-cpp` to `pyarrow`. Do you mean `/nvme/0/pgali/envs/cudfdev/include/arrow/python/platform.h` should still exist? I feel it is a path that corresponds to `arrow-cpp`(since it says `arrow` & not `pyarrow`).

Reply:
@robertmaynard for wheel packaging (as opposed to conda) everything needs to be self contained in the `site-packages` directory, so it couldn't land in a place where CMake would naturally search.

In theory this could be handled differently for the conda packages, but it would likely be a non-trivial amount of work and maintenance.

Reply:
Thanks @galipremsagar and @kkraus14 for the explanations. If this is the layout that makes the most sense from a python package perspective we will have to live with it.

Reply:
This is one of the fundamental problems for pip packaging wherein both library dependencies and CMake configs don't get installed into any globally visible places. Not much we can do here, given the limitations I think it makes sense for pyarrow not to diverge in its wheel and conda packaging. FWIW this approach has precedent from packages like numpy, and cudf already uses this exact logic to find arrow when we build wheels against the libarrow contained in the pyarrow wheel.